viewer/packages/camera-manager/src/CameraManager.ts

```diff
@@ -1,604 +1,68 @@
-    cameraPosition: THREE.Vector3,
-    cameraDirection: THREE.Vector3,
-    corners: Array<THREE.Vector3>
-  ): number {
-    const { nearPlane, nearPlaneCoplanarPoint } = this._calculateCameraFarBuffers;
-
-    nearPlaneCoplanarPoint.copy(cameraPosition).addScaledVector(cameraDirection, near);
-    nearPlane.setFromNormalAndCoplanarPoint(cameraDirection, nearPlaneCoplanarPoint);
-    let far = -Infinity;
-    for (let i = 0; i < corners.length; ++i) {
-      if (nearPlane.distanceToPoint(corners[i]) >= 0) {
-        const dist = corners[i].distanceTo(cameraPosition);
-        far = Math.max(far, dist);
-      }
-    }
-    far = Math.max(near * 2, far);
-
-    return far;
-  }
-
-  private calculateCameraNear(
-    camera: THREE.PerspectiveCamera,
-    combinedBbox: THREE.Box3,
-    cameraPosition: THREE.Vector3
-  ): number {
-    let near = combinedBbox.distanceToPoint(cameraPosition);
-    near /= Math.sqrt(1 + Math.tan(((camera.fov / 180) * Math.PI) / 2) ** 2 * (camera.aspect ** 2 + 1));
-    near = Math.max(0.1, near);
-
-    return near;
-  }
-
+  getCamera(): THREE.PerspectiveCamera;
   /**
-   * Calculates new target when raycaster doesn't have any intersections with the model.
-   * @param cursorPosition Cursor position for desired calculations.
-   * @param cursorPosition.x
-   * @param cursorPosition.y
+   * Set camera's state. Rotation and target can't be set at the same time. as they could conflict.
+   *
+   * @param state Camera state, all fields are optional.
+   * @param state.position Camera position in world space.
+   * @param state.target Camera target in world space.
+   * @param state.rotation Camera local rotation in quaternion form.
+   * @example
+   * ```js
+   * // store position, target
+   * const { position, target } = cameraManager.getCameraState();
+   * // restore position, target
+   * cameraManager.setCameraState({ position, target });
+   * ```
    */
-  private calculateNewTargetWithoutModel(
-    cursorPosition: { x: number; y: number },
-    modelsBoundingBox: THREE.Box3
-  ): THREE.Vector3 {
-    const modelSize = modelsBoundingBox.min.distanceTo(modelsBoundingBox.max);
-
-    const lastScrollTargetDistance = this.cameraControls.getScrollTarget().distanceTo(this._camera.position);
-
-    const newTargetDistance =
-      lastScrollTargetDistance <= this.cameraControls.minDistance
-        ? Math.min(this._camera.position.distanceTo(modelsBoundingBox.getCenter(new THREE.Vector3())), modelSize) / 2
-        : lastScrollTargetDistance;
-
-    this._raycaster.setFromCamera(cursorPosition, this._camera);
-
-    const farPoint = this._raycaster.ray.direction
-      .clone()
-      .normalize()
-      .multiplyScalar(newTargetDistance)
-      .add(this._camera.position);
-
-    return farPoint;
-  }
+  setCameraState(state: CameraState): void;
 
   /**
-   * Calculates new camera target based on cursor position.
-   * @param event MouseEvent that contains pointer location data.
+   * Get camera's state
    */
```

Comment:
Add doc for each of the fields in `CameraState`, e.g.
```
@param state.position The camera position in world coordinates
```

Reply:
Are you sure we want `@param` tag even on return? I added it as: 
`@returns Camera state: position, target and rotation.
`

Reply:
Also, returning type describes how state looks, so I think it's ok to have such description